src/main/kotlin/uk/gov/dwp/dataworks/logging/LoggerLayoutAppender.kt

```diff
@@ -0,0 +1,35 @@
+package uk.gov.dwp.dataworks.logging
+
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.core.LayoutBase
+
+/**
+ * Implementation of [LayoutBase] which allows us to parse events and reformat into JSON structured messages.
+ */
+object LoggerLayoutAppender : LayoutBase<ILoggingEvent>() {
+
+    private var start_time_milliseconds = System.currentTimeMillis()
+
+    private fun getDurationInMilliseconds(epochTime: Long): String {
+        val elapsedMilliseconds = epochTime - start_time_milliseconds
+        return elapsedMilliseconds.toString()
+    }
+
+    override fun doLayout(event: ILoggingEvent?): String {
+        if (event == null) {
+            return ""
+        }
+        return """
+            |{
+                |"timestamp":"${epochToUTCString(event.timeStamp)}",
+                |"log_level":"${event.level}",
+                |"message":${flattenString(event.formattedMessage)},
+                |"exception":"${throwableProxyEventToString(event)}",

```

Comment:
The Exception tuple should be conditional on there being an exception, as this is what all standard loggers do.

Reply:
Yes, I missed this nuance in my migration. Please see [above comment](https://github.com/dwp/dataworks-common-logging/pull/1#discussion_r390174070)